### PR TITLE
email/pw auth integration with firestore db

### PR DESCRIPTION
**Pull request by devintor**
barebones site but fully functional with email and password auth, error handling, routing, etc.

updates user info in firestore database

screen recording:
https://github.com/devintor/breadbox/assets/139282865/f324659b-7a96-484d-9219-9b49b79d8a16

### CODE REVIEW

**Comment by devintor** [comment]
Currently have everything related to GoogleAuth commented out because it has some bugs and I just wanted to show the email/password setup for now

File: backend/src/server.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const middleware = require('./middleware');
+
+const app = express();
+const port = 3000;
+
+app.use(cors());
+
+app.use(middleware.decodeToken);
+
+app.get('/api/todos', (req, res) => {
```
Comment (ecristerna) [inline backend/src/server.js:12]: which feature does this task/todos map to from the project plan?

File: backend/src/server.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const express = require('express');
```
Comment (ecristerna) [inline backend/src/server.js:1]: let's find a better name for this file

File: frontend/src/components/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      await signInWithEmailAndPassword(auth, email, password);
+      console.log("User logged in Successfully");
```
Comment (ecristerna) [inline frontend/src/components/Login.jsx:15]: rm

File: frontend/src/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      window.location.href = "/profile";
+      toast.success("User logged in Successfully", {
+        position: "top-center",
+      });
+    } catch (error) {
+      console.log(error.message);
```
Comment (ecristerna) [inline frontend/src/components/Login.jsx:21]: rm

File: frontend/src/components/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+      toast.error(error.message, {
```
Comment (ecristerna) [inline frontend/src/components/Login.jsx:23]: is there a way to show the human readable string instead? 
Screenshot 2024-07-08 at 09 06 04

File: frontend/src/components/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+      <div className="d-grid">
+        <button type="submit" className="btn btn-primary">
+          Submit
+        </button>
+      </div>
+      <p className="forgot-password text-right">
+        New user <a href="/register">Register Here</a>
+      </p>
+      {/* <GoogleAuth/> */}
```
Comment (ecristerna) [inline frontend/src/components/Login.jsx:63]: rm

File: frontend/src/components/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          onChange={(e) => setPassword(e.target.value)}
+        />
+      </div>
+
+      <div className="d-grid">
+        <button type="submit" className="btn btn-primary">
+          Submit
+        </button>
+      </div>
+      <p className="forgot-password text-right">
```
Comment (ecristerna) [inline frontend/src/components/Login.jsx:60]: rename this class?

File: frontend/src/components/Profile.jsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+              src={userDetails.photo}
+              width={"40%"}
+              style={{ borderRadius: "50%" }}
+            />
+          </div>
+          <h3>Welcome {userDetails.firstName} 🙏🙏</h3>
+          <div>
+            <p>Email: {userDetails.email}</p>
+            <p>First Name: {userDetails.firstName}</p>
+            {/* <p>Last Name: {userDetails.lastName}</p> */}
```
Comment (ecristerna) [inline frontend/src/components/Profile.jsx:49]: rm

File: frontend/src/components/Register.jsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+      const user = auth.currentUser;
+      console.log(user);
+      if (user) {
+        await setDoc(doc(db, "Users", user.uid), {
+          email: user.email,
+          firstName: fname,
+          lastName: lname,
+          photo:""
+        });
+      }
```
Comment (ecristerna) [inline frontend/src/components/Register.jsx:26]: can you explain this process? is `currentUser` the newly created user, or an existing one? then what are we doing with the `setDoc()` call?

**Review by ecristerna — APPROVED** [review]
Let's remove all the commented code, and file that are not needed/used today, before merging this to main. If you want to keep the Google OAuth code, you can fork this branch and create a new one, but do not merge code and files (and images) into main that are not actually used.

Also, please make sure there are no `console.log` in your code before submitting.

**Pull request merged**